network.go

```diff
@@ -195,6 +195,12 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 	if name == "" {
 		return nil, ErrInvalidName(name)
 	}
+
+	e, _ := n.EndpointByName(name)
```

Comment:
am talking about code readability here.
It will read as : if there is no error in EndpointByName method, then throw an error.
While with `if e != nil`, it will read as if there is an existing endpoint, then throw an error.

Anyways, if that is what everyone prefers, I can change it. But IMHO, your suggestion makes it less readable.